Lib/test/libregrtest/results.py

```diff
@@ -119,11 +122,17 @@ def accumulate_result(self, result: TestResult, runtests: RunTests):
             self.stats.accumulate(result.stats)
         if rerun:
             self.rerun.append(test_name)
-
+        if result.covered_lines:
+            # we don't care about trace counts so we don't have to sum them up
+            self.covered_lines.update(result.covered_lines)
         xml_data = result.xml_data
         if xml_data:
             self.add_junit(xml_data)
 
+    def get_coverage_results(self) -> trace.CoverageResults:
+        counts = {loc: 1 for loc in frozenset(self. covered_lines)}
```

Comment:
An extra space here. Also is there a reason to convert the results to `frozenset`? Even though the coverage is still going, the data in `self.covered_lines` is not going to change right? I'm assuming this could be a large set, what's the benefit to basically copy it here?

Reply:
Good catch, this is unnecessary. In an earlier version of this, I used more complicated coverage agreegation that happened on every WorkerThread, and as such, I needed to make sure the counts wouldn't change during iteration. Now `get_coverage_results()` is only called after all worker runs finished.

As for the performance, I wondered if that would indeed be an issue, and you're right there as well. It takes 47 seconds on my Mac to convert a set with a million tuples with unique strings and integers inside. CPython currently includes around 550k non-whitespace lines of Python code.

Reply:
Wait, no, `timeit` doesn't automatically divide by `number`. A single call to convert a set of a million tuples into a frozenset is 47 milliseconds, which I guess is insignificant in the grand scheme of a test suite.

Reply:
Ah, that's actually faster than I expected, probably because it knows the size of the memory required and only need to allocate it once.